### ISSUE TITLE
🚀 Feature: lifecycle state for Templates

### ISSUE THREAD

**Issue by david-christ** [backstage/backstage#27922]
### 🔖 Feature description

Currently, `kind: Template` does not allow for a field to define what stage of development the template is in.

### 🎤 Context

We would like for certain templates to only be available to e.g. beta testers. 

```ts
        <ScaffolderPage
          templateFilter={template => {
            const featureFlagsApi = useApi(featureFlagsApiRef);
            const isFeatureFlagToggled = featureFlagsApi.isActive('enable-beta-templates');
            const isProd = template.SOME_PROPERTY && template.SOME_PROPERTY === 'production';
            return isFeatureFlagToggled || isProd;
          }}
        />
```

We could (mis-)use an existing field like `metadata.tags` or `metadata.annotations`, but that does not feel right (and in case of tags bleeds through to the UI layer easily.

### ✌️ Possible Implementation

Use `spec.lifecycle`, like in other entity `kind`s.

Not sure whether it would be enough to add the field to `plugins/scaffolder-common/src/TemplateEntityV1beta3.ts`?

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by benjdlambert** [comment]
Yeah, I think there's also some types in the  `scaffolder-common` too https://github.com/backstage/backstage/blob/88746203aa6c7b171d381896bc07d77b11d44b1e/plugins/scaffolder-common/src/TemplateEntityV1beta3.ts#L31

**Comment by benjdlambert** [comment]
ping @tylerd-canva @david-christ did any of you manage to take a look at this? :pray:

**Comment by tylerd-canva** [comment]
Finally had some time. Let me know if there's any more to it.
